psydac/ddm/cart.py

```diff
@@ -1124,110 +1138,142 @@ def __init__(self, npts, pads, periods, comm, shifts, axes, exts, ranks_in_topo,
 
     @property
     def ranks_in_topo_minus( self ):
+        """ the ranks in the cartesian topology of defined on the minus patch."""
         return self._ranks_in_topo_minus
 
+    @property
+    def ranks_in_topo_plus( self ):
+        """ the ranks in the cartesian topology of defined on the plus patch."""
+        return self._ranks_in_topo_plus
+
     @property
     def coords_from_rank_minus( self ):
+        """ Array that maps the ranks of minus patch to their coordinates in the cartesian decomposition."""
         return self._coords_from_rank_minus
 
     @property
     def coords_from_rank_plus( self ):
+        """ Array that maps the ranks of plus patch to their coordinates in the cartesian decomposition."""
         return self._coords_from_rank_plus
 
     @property
     def boundary_ranks_minus( self ):
+        """ Array that contains the ranks defined on the boundary of the minus side of the interface."""
         return self._boundary_ranks_minus
 
     @property
     def boundary_ranks_plus( self ):
+        """ Array that contains the ranks defined on the boundary of the plus side of the interface."""
         return self._boundary_ranks_plus
 
     @property
     def reduced_global_starts_minus( self ):
+        """ The starts of all the processes in the cartesian decomposition of the minus patch with the shift=1"""
         return self._reduced_global_starts_minus
 
     @property
     def reduced_global_starts_plus( self ):
+        """ The starts of all the processes in the cartesian decomposition of the plus patch with the shift=1"""
         return self._reduced_global_starts_plus
 
     @property
     def reduced_global_ends_minus( self ):
+        """ The ends of all the processes in the cartesian decomposition of the minus patch with the shift=1"""
         return self._reduced_global_ends_minus
 
     @property
     def reduced_global_ends_plus( self ):
+        """ The ends of all the processes in the cartesian decomposition of the plus patch with the shift=1"""
         return self._reduced_global_ends_plus
 
     @property
     def global_starts_minus( self ):
+        """ The starts of all the processes in the cartesian decomposition defined on the minus patch."""
         return self._global_starts_minus
 
     @property
     def global_starts_plus( self ):
+        """ The starts of all the processes in the cartesian decomposition defined on the plus patch."""
         return self._global_starts_plus
 
     @property
     def global_ends_minus( self ):
+        """ The ends of all the processes in the cartesian decomposition defined on the minus patch."""
         return self._global_ends_minus
 
     @property
     def global_ends_plus( self ):
+        """ The ends of all the processes in the cartesian decomposition defined on the plus patch."""
         return self._global_ends_plus
 
     @property
     def axis( self ):
+        """ The axis of the interface."""
         return self._axis
```

Comment:
I do not fully understand this. The interface could be represented by different axis on the two sides, so we should have `axis_plus` and `axis_minus` I think.

Reply:
yes, you are right, in our case, I assume they are the same, once we generalize it we will have two different axis

Reply:
But even for the pretzel domain the axes are not always the same on the two sides of the interface... How do you make that work?

Reply:
The extremities are different but the axis is the same in both sides, it's easy to do add an affine mapping to impose that the axis are the same

Reply:
I had not realized that the pretzel was such a lucky geometry!

As for the trick of using an affine mapping, we discussed about this before. The counter-example is the simple situation where 3 square patches converge into one vertex.

Reply:
no, I have changed it, the first time we had it, the axis was different and I changed it to be the same, you can easily do that with any geometry by adding an affine mapping.
As for your counter-example, I'm not sure I can see why it doesn't work
that example was not working for different reasons

Reply:
It does not work because you have 3 patches (0, 1, 2) and 3 interfaces, with the following connectivity:
```python
connectivity = {
    (0, 1) : ((0, +1), (0, -1)),   # OK
    (1, 2) : ((1, +1), (1, +1)),   # OK
    (2, 0) : ((0, +1), (1, +1)),   # ISSUE!!
}
```

After drawing it on a piece of paper, one can quickly convince oneself that there is no way to avoid a change of axis in one of the interfaces.

Reply:
ah yes, you are right, it won't work, we can easily fix it, I will do it after the next PR

Reply:
I was thinking about the direction, and I confused it with the axis, we can change the direction however we want but not the axis

Reply:
I have created issue #236 for this